### DISCUSSION TITLE
Performance recommendations for SSD's

### DISCUSSION THREAD

**Discussion by TheScriptGuy** [syslog-ng/syslog-ng#4266] [Q&A] [answered]
Hi!

I need some guidance please when it comes to performance recommendations when the output destination file is on a SSD (specifically around minimizing individual write events to disk). When searching on the internet, I keep getting referred to the configuration/administration guides, but they just talk about which options to use and have limited examples of how to use the commands.

Here's what I'm trying to do:
I have a syslog-ng server that receives udp syslog messages from multiple hosts. I'd ideally like to minimize the number of write  events to file on my SSD (i.e. write in bulk, not as each individual message comes in). I'm not concerned about message latency or if messages get lost (e.g. power failure). 

I stumbled on the `batch-lines()` parameter but found out that it only applies to HTTP destinations.
I eventually found the `flush-lines()` parameter that can be used in conjunction with file and set it to a 1000. When I tail my log files though, it still looks like they're being written in a "streaming" manner.

I'm sure I'm missing something trivial here, but I'm at a loss as to what it might be. Any guidance appreciated!

Here's a portion of my syslog-ng.conf:
```
options {
  create_dirs(yes);
  mark_freq(3600);
  stats_freq(43200);
  time_reopen(5);
  use_dns(no);
  dns-cache(no);
  owner(root);
  group(adm);
  perm(0640);
  dir_perm(0755);
};
source s_net { udp(ip(0.0.0.0) port(514)); };
destination d_net_syslog { file("/syslog/$HOST/device.$YEAR-$MONTH-$DAY" flush-lines(10000)); };
log { source(s_net); destination(d_net_syslog); flags(final); };
```

**Comment by bazsi** [comment, marked as answer]
Syslog-ng will automatically batch writes as long as the input rate is high
enough.

E.g. it will collect up-to flush-lines() number of messages and write it
via a single writev() system call. But if the source is not producing
enough messages and the destination becomes "idle" it will just flush the
output.

So whenever you are testing with a message you will see that syslog-ng will
write individual messages.

But if you produce a steady rate, the number of messages flushed at one
time will increase.

If you look at syscall statistics, you should see that syslog-ng would
write the file using writev() the number of iov elements the number of
messages flushed at the same time.

This approach automatically optimizes for both throughput and latency.
